bndtools.m2e/src/bndtools/m2e/MavenRunProvider.java

```diff
@@ -136,4 +139,13 @@ private Bndrun create0(IResource targetResource, IMavenProjectFacade projectFaca
 		}
 	}
 
+	@Reference
+	public void setMavenWorkspaceRepository(MavenWorkspaceRepository mavenWorkspaceRepository) {
+		this.mavenWorkspaceRepository = mavenWorkspaceRepository;
+	}
+
+	public void unsetMavenWorkspaceRepository(MavenWorkspaceRepository mavenWorkspaceRepository) {
+		this.mavenWorkspaceRepository = null;
```

Comment:
Isn't the correct thing:

```java
if (this.mavenWorkspaceRepository == mavenWorkspaceRepository) {
  this.mavenWorkspaceRepository = null;
}
```

to properly handle replacement?

Reply:
fixed.